applications/MetisApplication/custom_processes/metis_divide_submodelparts_heterogeneous_input_process.cpp

```diff
@@ -0,0 +1,110 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Daniel Diez
+//
+
+
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "custom_processes/metis_divide_submodelparts_heterogeneous_input_process.h"
+
+namespace Kratos {
+
+
+void MetisDivideSubModelPartsHeterogeneousInputProcess::GetNodesPartitions(
+    std::vector<idxtype> &rNodePartition,
+    SizeType &rNumNodes) {
+
+        KRATOS_TRY
+
+        SizeType num_nodes_in_mesh = BaseType::mrIO.ReadNodesNumber();
+        rNodePartition.resize(num_nodes_in_mesh);
+
+        for (auto submodelpart_name : mSettings["submodelpart_list"].GetStringArray()) {
+            // Read nodal graph from input
+            IO::ConnectivitiesContainerType kratos_format_node_connectivities;
+            std::unordered_set<SizeType> submodelpart_elements_ids;
+            std::unordered_set<SizeType> submodelpart_condtions_ids;
+            std::vector<idxtype> submodelpart_partition_nodes;
+
+            BaseType::mrIO.ReadSubModelPartElementsAndConditionsIds(
+                submodelpart_name,
```

Comment:
I am pretty sure this doesn't work with SubSubModelParts
=> #7108 should fix it, for me is ok to wait

Reply:
There is a test working with sub-sub model parts. Remember we are only working with mdpa file, there are no ModelParts yet at this moment, so I dont see how #7108 would affect this. The `ReadSubModelPartElementsAndConditionsIds` only looks for the given model part name in the mdpa.